src/gui/screen_printing_serial.cpp

```diff
@@ -131,28 +156,25 @@ int screen_printing_serial_event(screen_t *screen, window_t *window, uint8_t eve
     if (status_footer_event(&(pw->footer), window, event, param)) {
         return 1;
     }
-
-    static char lock = 0;
-    if (lock)
+    if (event != WINDOW_EVENT_CLICK) {
         return 0;
-    lock = 1;
-
-    if (event == WINDOW_EVENT_BTN_DN) {
-        screen_open(get_scr_menu_tune()->id);
-        lock = 0;
-        return 1; //here MUST BE return 1. Screen is no longer valid.
     }
 
-    int now = HAL_GetTick();
-
-    if ((now - pw->last_tick) > 500) {
-        pw->last_tick = now;
-        static int i = 0;
-        ++i;
-        i %= (sizeof(txts) / sizeof(txts[0]));
-        window_set_text(pw->w_message.win.id, txts[i]);
+    switch (((int)param) - 1) {
```

Comment:
OMG, win32 bullsh*t sending unspecified 4 bytes to handler functions. I really hate this reinterpret_cast of the `param` parameter.

Reply:
param - void* will be obsolete, I hope, in refactored version of screens / windows - now used reinterpret_cast<int>